scripts/zkai_native_attention_mlp_transcript_stable_comparison_gate.py

```
@@ -0,0 +1,870 @@
+        isinstance(value, str)
+        and value.startswith(prefix)
+        and len(value) == len(prefix) + 64
+        and all(char in "0123456789abcdef" for char in value[len(prefix) :])
+    )
+
+
+def read_json(path: pathlib.Path, label: str) -> tuple[dict[str, Any], bytes]:
+    resolved = path.resolve()
+    evidence_root = EVIDENCE_DIR.resolve()
+    if evidence_root not in resolved.parents and resolved != evidence_root:
+        raise TranscriptStableComparisonError(f"{label} path must stay under docs/engineering/evidence")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)